variable/variable.cpp

```diff
@@ -15,198 +15,133 @@
-  return *this;
+void Variable::expectCanSetUnit(const units::Unit &unit) const {
+  if (this->unit() != unit && is_slice())
+    throw except::UnitError("Partial view on data of variable cannot be used "
+                            "to change the unit.");
 }
 
-template <class T> VariableView VariableView::assign(const T &other) const {
-  if (*this == VariableConstView(other))
-    return *this; // Self-assignment, return early.
-  underlying().data().copy(other, *this);
-  return *this;
+void Variable::setUnit(const units::Unit &unit) {
+  expectCanSetUnit(unit);
+  m_object->setUnit(unit);
 }
 
-template SCIPP_VARIABLE_EXPORT VariableView
-VariableView::assign(const Variable &) const;
-template SCIPP_VARIABLE_EXPORT VariableView
-VariableView::assign(const VariableConstView &) const;
-template SCIPP_VARIABLE_EXPORT VariableView
-VariableView::assign(const VariableView &) const;
-
-bool VariableConstView::operator==(const VariableConstView &other) const {
+bool Variable::operator==(const Variable &other) const {
   if (!*this || !other)
     return static_cast<bool>(*this) == static_cast<bool>(other);
-  // Always use deep comparison (pointer comparison does not make sense since we
-  // may be looking at a different section).
-  return underlying().data().equals(*this, other);
+  // Note: Not comparing strides
+  return dims() == other.dims() && data().equals(*this, other);
 }
 
-bool VariableConstView::operator!=(const VariableConstView &other) const {
+bool Variable::operator!=(const Variable &other) const {
   return !(*this == other);
 }
 
-void VariableView::setUnit(const units::Unit &unit) const {
-  expectCanSetUnit(unit);
-  m_mutableVariable->setUnit(unit);
-}
-
-void VariableView::expectCanSetUnit(const units::Unit &unit) const {
-  if ((this->unit() != unit) && (dims() != m_mutableVariable->dims()))
-    throw except::UnitError("Partial view on data of variable cannot be used "
-                            "to change the unit.");
-}
-
-VariableConstView Variable::slice(const Slice slice) const & {
-  return {*this, slice.dim(), slice.begin(), slice.end()};
-}
-
-Variable Variable::slice(const Slice slice) const && {
-  return Variable{this->slice(slice)};
-}
-
-VariableView Variable::slice(const Slice slice) & {
-  return {*this, slice.dim(), slice.begin(), slice.end()};
-}
-
-Variable Variable::slice(const Slice slice) && {
-  return Variable{this->slice(slice)};
-}
-
-VariableConstView VariableConstView::slice(const Slice slice) const {
-  return VariableConstView(*this, slice.dim(), slice.begin(), slice.end());
-}
-
-VariableView VariableView::slice(const Slice slice) const {
-  return VariableView(*this, slice.dim(), slice.begin(), slice.end());
+scipp::span<const scipp::index> Variable::strides() const {
+  return {m_strides.begin(), dims().ndim()};
+}
+
+core::ElementArrayViewParams Variable::array_params() const noexcept {
+  // TODO Translating strides into dims, which get translated back to
```

Comment:
Should this be addressed in this PR? Or a ticket be created so that this "TODO" doesn't get lost?

Reply:
I think we will do it separately.